File: typescript/apps/app3/clsdir/business_person.ts
import { Person } from './person.ts';

export class BusinessPerson implements Person {
    private name: string;
    private age: number;
    protected address: string;
    private country: string;
    private job: string;


    public constructor(name:string, age:number,
        address:string, country:string, job:string) {
        this.name = name;
        this.age = age;
        this.address = address;
        this.country = country;
        this.job = job;
    }

    public introduce(): void {
        console.log("Nice to meet you.");
        console.log("My name is " + this.name +  ". I'm " + this.age + " years old.");
        console.log("I'm now live in " + this.country + ", at " + this.address + ". ");
        console.log("I'm work as " + this.job);
    } 

    public getName(): string {
        return this.name;
    }
    public getAge(): number {
        return this.age;
    }
    public getAddress(): string {
        return this.address;
    }
    public getCountry(): string {
        return this.country;
    }
}

File: csharp/apps/app2/cls/BusinessPerson.cs
public class BusinessPerson : Person {
    private string job {set; get;}
    private string name { get; set; }
    private int age { get; set; }
    private string country { get; set; }

    public BusinessPerson(string name, int age, string country
    , string job) {
        this.name = name;
        this.age = age;
        this.country = country;
        this.job = job;
    }

    public void sayHello() {
        Console.WriteLine("Hello, my name is " + this.name + "." );
        Console.WriteLine("I'm " + this.age + " years old.");
        Console.WriteLine("I'm work as " + this.job + " in " + this.country + " now.");
    }
}